src/reflect/scala/reflect/internal/tpe/GlbLubs.scala

```diff
@@ -544,7 +544,7 @@ private[internal] trait GlbLubs {
     }
     // if (settings.debug.value) { println(indent + "glb of " + ts + " at depth "+depth); indent = indent + "  " } //DEBUG
     if (StatisticsStatics.areSomeColdStatsEnabled) statistics.incCounter(nestedLubCount)
-    glb0(ts)
+    glb0(ts.map(_.dealias))
```

Comment:
I've been poking around a bit here. The dealias is needed because `typeDepth` considers type aliases to have depth 1, regardless of their RHS. Trying to figure out whether we can avoid this `dealias`, but no luck so far